evaluate/report/collection.go

```diff
@@ -86,10 +86,13 @@ func (a AssessmentPerModelPerLanguagePerRepository) CollapseByModel() Assessment
 }
 
 // WalkByScore walks the given assessment metrics by their score.
-func WalkByScore(assessmentsPerModel map[string]metrics.Assessments, function func(model string, assessment metrics.Assessments, score uint) error) error {
+func WalkByScore(assessmentsPerModel AssessmentPerModel, function func(model model.Model, assessment metrics.Assessments, score uint) error) error {
 	models := maps.Keys(assessmentsPerModel)
-	sort.Strings(models)
-	scores := make(map[string]uint, len(models))
+	sort.SliceStable(models, func(i, j int) bool {
+		return models[i].ID() < models[j].ID()
+	})
```

Comment:
Would use the following to avoid dealing with indices:

```go
slices.SortStableFunc(models, func(a, b model.Model) int {
	return cmp.Compare(a.ID(), b.ID())
})
```